specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json

```diff
@@ -4701,8 +4828,17 @@
       "x-ms-discriminator-value": "Scheduled"
     },
     "ScheduledAlertRuleProperties": {
-      "description": "Alert rule property bag.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/ScheduledAlertRuleCommonProperties"
+        }
+      ],
+      "description": "Scheduled alert rule base property bag.",
       "properties": {
+        "alertRuleTemplateName": {
```

Comment:
Why is this just a name and not a full resourcd ID?  If referencing another resource its much safer to use the full ID.

Reply:
Well,
the AlertRuleTemplate resource is a read only resource that all our customers have by default. every AlertRuleTemplate resource has a predefined unique resource name (In fact, this is a Guid). 
When users call the GetAll endpoint of AlertRuleTemplates, they receive all the templates resources, and the resource unique name appears under the "Name" property (like any other ARM resource). that's why we called it "alertRuleTemplateName".
we **don't need the full id** of the template resource (e.g. "subsciprtions/.../resourcegroups/.../.../alertRuleTemplate/[guid]"), because we only need the template Name, and it is the same for all subscriptions, rgs, etc.

Reply:
Is the template tied to a workspace?  I.e. the "status" of the template is unique to the workspace?  

Does "alertRulesCreatedByTemplateCount" increment across workspaces?  I.e. if I create an alertRule from a template in workspace foo then GET the template in workspace bar will "alertRulesCreatedByTemplateCount" be 1?

Reply:
This is borderline.  In my opinion any resource referencing should be by full resource ID.  However, if you want people to easily transfer alertRules from workspace to workspace that would be onerous as they would have to update the resource ID.  I ask the above questions because alertRuleTemplates could technically be a tenant level readonly resource on your RP if they don't have any workspace-specific state.  Similar to how built-in roleDefinitions and policyDefinitions work in Microsoft.Authorization.

If they do have state or copying alertRules (based on a template) across workspaces is a useful scenario then this is fine.

Reply:
Yes, when calling "GET" on AlertRuleTemplate, the user gets also the status of the template upon the workspace, and alertRulesCreatedByTemplateCount is indicating how many rules was created with the template. So it can't be a tenant resource.
And yes, we would like the usage of templates to be as simple as possible, and allow user to PUT the same rule on different workspaces without having to change the template ID according to the workspace. We don't need the full id of the template in our backend to create a rule, and it has no value for the users, imo.

We would like to keep the definition as is, please 🙂
It was PM-approved a month ago and already working seamlessly in private preview :)